src/components/Graph.react.js

```diff
@@ -69,6 +69,19 @@ export default class PlotlyGraph extends Component {
         super(props);
         this.bindEvents = this.bindEvents.bind(this);
         this._hasPlotted = false;
+
+        this.props.id = this.props.id ? this.props.id : this.generateId();
```

Comment:
This looks wrong. A component should never override its props.

If using state, setProps or another approach brakes tests, maybe we need to dig and find if that is correct or incorrect behavior from either the test or sub-components.

Reply:
I agree that it's wrong. 

However, seeing as the test do indeed break, I'm not sure how to go from here. There are no sub-components, and the tests seem correct. Any help would be appreciated!